feat: add `math/base/special/tribonaccif` (#6066)
stdlib-js/stdlib: lib/node_modules/@stdlib/math/base/special/tribonaccif/README.md

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

> # Tribonaccif

Reviewer: ```suggestion
# tribonaccif
```

Either needs to be the alias or a title. E.g., here either "Tribonacci" (title) or an alias "tribonaccif". No mixing the two.

Reviewer 2: Noted! Thank you!

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# Tribonaccif

> > Compute the nth [Tribonacci number][tribonacci-number] for [single-precision floating-point number][ieee754].

Reviewer 3: ```suggestion
> Compute the nth [Tribonacci number][tribonacci-number] as a [single-precision floating-point number][ieee754].
```

    <br>
</div> -->

<!-- </equation> -->

The sequence is defined by the recurrence relation

<!-- <equation class="equation" label="eq:tribonacci_recurrence_relation" align="center" raw="F_n = F_{n-1} + F_{n-2} + F_{n-3}" alt="Tribonacci sequence recurrence relation"> -->

```math
F_n = F_{n-1} + F_{n-2} + F_{n-3}
```

<!-- <div class="equation" align="center" data-raw-text="F_n = F_{n-1} + F_{n-2} + F_{n-3}" data-equation="eq:tribonacci_recurrence_relation">
    <img src="https://cdn.jsdelivr.net/gh/stdlib-js/stdlib@3249a68fb57cd71148f87ef3b2774be70a04d80a/lib/node_modules/@stdlib/math/base/special/tribonacci/docs/img/equation_tribonacci_recurrence_relation.svg" alt="Tribonacci sequence recurrence relation">
    <br>
</div> -->

<!-- </equation> -->

with seed values `F_0 = 0`, `F_1 = 0`, and `F_2 = 1`.

</section>

<!-- /.intro -->

<section class="usage">

## Usage

```javascript
var tribonaccif = require( '@stdlib/math/base/special/tribonaccif' );
```

#### tribonaccif( n )

> Computes the nth [Tribonacci number][tribonacci-number] for [single-precision floating-point number][ieee754].

Reviewer 3: ```suggestion
Computes the nth [Tribonacci number][tribonacci-number] as a [single-precision floating-point number][ieee754].
```

Reviewer 3: @Neerajpathak07 Can you update this here and elsewhere? Because, we are not pointing `n` to be in single-precision, but we are referring to the nth tribonacci number to be in single-precision.

Author: sure

```javascript
var v = tribonaccif( 0 );
// returns 0

v = tribonaccif( 1 );
// returns 0

v = tribonaccif( 2 );
// returns 1

v = tribonaccif( 3 );
// returns 1

v = tribonaccif( 30 );
// returns 15902591
```

If `n > 30`, the function returns `NaN`, as larger [Tribonacci numbers][tribonacci-number] cannot be safely represented in [single-precision floating-point format][ieee754].

```javascript

> var v = tribonaccif( 32 );

Reviewer 3: ```suggestion
var v = tribonaccif( 31 );
```
Even though 32 is also fine, let's keep the closest value to the limit of 30, i.e., 31 here, similar to what we have in the double-precision equivalent, `tribonacci`.

Author: Ok np I'll update it in the `repl` also.

// returns NaN
```

If not provided a nonnegative integer value, the function returns `NaN`.

```javascript
var v = tribonaccif( 3.14 );
// returns NaN

v = tribonaccif( -1 );
// returns NaN
```

If provided `NaN`, the function returns `NaN`.

```javascript
var v = tribonaccif( NaN );
// returns NaN
```

</section>

<!-- /.usage -->

<section class="notes">

</section>

<!-- /.notes -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var tribonaccif = require( '@stdlib/math/base/special/tribonaccif' );

var v;
var i;

for ( i = 0; i < 31; i++ ) {
    v = tribonaccif( i );
    console.log( v );
}
```

</section>

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/math/base/special/tribonaccif.h"
```

#### stdlib_base_tribonaccif( n )

> Computes the nth [Tribonacci number][tribonacci-number] [single-precision floating-point number][ieee754].

Reviewer 3: ```suggestion
Computes the nth [Tribonacci number][tribonacci-number] [single-precision floating-point number][ieee754].
```
This must be updated.

```c

> float out = stdlib_base_tribonaccif( 0f );
> // returns 0f
> 
> out = stdlib_base_tribonaccif( 1f );
> // returns 0f

Reviewer 3: ```suggestion
float out = stdlib_base_tribonaccif( 0 );
// returns 0

out = stdlib_base_tribonaccif( 1 );
// returns 0
```

Reviewer 3: @Neerajpathak07 We don't need to add the `f` suffixes after integers here.

```c
float stdlib_base_tribonaccif( const int32_t n );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/math/base/special/tribonaccif.h"
#include <stdio.h>
#include <stdint.h>

int main( void ) {
    int32_t i;
    float v;

    for ( i = 0; i < 31; i++ ) {
        v = stdlib_base_tribonaccif( i );
        printf( "tribonaccif(%d) = %f\n", i, v );
    }
}
```

</section>

<!-- /.examples -->

</section>

<!-- /.c -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->

<section class="related">

</section>

<!-- /.related -->

<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="links">

[tribonacci-number]: https://en.wikipedia.org/wiki/Generalizations_of_Fibonacci_numbers#Tribonacci_numbers

[ieee754]: https://en.wikipedia.org/wiki/IEEE_754-1985

> 

Reviewer 3: ```suggestion

<!-- <related-links> -->

<!-- </related-links> -->

```

Reviewer 3: Missing section for related links.
